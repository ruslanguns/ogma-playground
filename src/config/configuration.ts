export default () => ({
  port: +process.env.PORT || 3000,
  env: process.env.NODE_ENV || 'development',
  applicationName: 'OgmaDemo',
  logging: {
    json: process.env.LOGS_IN_JSON_ENABLE === 'true',
    deleteFrecuency: +process.env.LOGS_DELETE_FREQUENCY_IN_HOURS || 96,
    disableExpressParser: process.env.LOGS_EXPRESS_PARSER_DISABLE === 'true',
  },
});
