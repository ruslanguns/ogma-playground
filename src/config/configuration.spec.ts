import configuration from './configuration';

process.env = {
  PORT: '1234',
  NODE_ENV: 'testing',
  LOGS_IN_JSON_ENABLE: 'true',
  LOGS_DELETE_FREQUENCY_IN_HOURS: '10',
  LOGS_EXPRESS_PARSER_DISABLE: 'true',
};

describe('Config Namespaces', () => {
  it('should match the snapshot', () => {
    const configValues = configuration();
    expect(configValues).toMatchSnapshot();
  });

  it('should logging delete frecuency', () => {
    const {
      logging: { deleteFrecuency },
    } = configuration();
    expect(deleteFrecuency).toEqual(10);
  });

  describe('Feature flags', () => {
    it('should enable or disable the logging json format', () => {
      const {
        logging: { json },
      } = configuration();
      expect(json).toEqual(true);
    });

    it('should enable or disable the logging json format', () => {
      const {
        logging: { disableExpressParser },
      } = configuration();
      expect(disableExpressParser).toEqual(true);
    });
  });
});
