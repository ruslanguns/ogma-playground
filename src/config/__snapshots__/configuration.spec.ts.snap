// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Namespaces should match the snapshot 1`] = `
Object {
  "applicationName": "OgmaDemo",
  "env": "testing",
  "logging": Object {
    "deleteFrecuency": 10,
    "disableExpressParser": true,
    "json": true,
  },
  "port": 1234,
}
`;
